### PR TITLE
SailfishOS calendar support

### PR DESCRIPTION
**Pull request by smurfy**
Adding Calendar support for sailfishOS
Adding Way to enable / disable calendar support using defines

@leppa Maybe you have time to have a quick look over the next couple of days.

### CODE REVIEW

File: fahrplan2.pro [hunk 1]
```diff
@@ -1,14 +1,13 @@
 VERSION = 2.0.24
 
+# Switch for jolla to separate harbour and openrepo version
+#DEFINES += BUILD_FOR_OPENREPOS
```
Comment (leppa) [inline fahrplan2.pro]: You can do something like:
```
openrepos {
    DEFINES += BUILD_FOR_OPENREPOS
}
```
and then run `qmake` with `CONFIG+=openrepos` to enable openrepos specific features.

File: src/main.cpp [hunk 2]
```diff
@@ -67,6 +69,13 @@ Q_DECL_EXPORT
 int main(int argc, char *argv[])
 {
     #if defined(BUILD_FOR_SAILFISHOS)
+        //To support calendar access
+        #if defined(BUILD_FOR_OPENREPOS)
+            qDebug()<<"openrepos.net build";
+            setuid(getpwnam("nemo")->pw_uid);
+            setgid(getgrnam("privileged")->gr_gid);
```
Comment (leppa) [inline src/main.cpp]: Is this really necessary when using invoker? If I'm not mistaken, invoker already runs as 'privileged' group.

File: fahrplan2.pro (hunk 1 above)
Comment (smurfy) [inline fahrplan2.pro]: ok, i usually compile using qt-creator (windows and so-on) but it makes sense to use it the way you suggested over removing/adding a comment.

File: src/main.cpp (hunk 2 above)
Comment (smurfy) [inline src/main.cpp]: i'm 99% sure :) I did several tests, including changing ```/usr/share/mapplauncherd/privileges``` but i think it was invoker + setgid which worked and was without modding a system-file. using setgid without the invoker does not work for example.
I will test it again tonight without using the setgid, but the invoker.

**Comment by smurfy** [comment]
Ok, i adapted the CONFIG solution, which is quite cool, i created a new release build configuration in qt-creator and now can easy switch between.

I also verified the setuid/setguid thing. It's needed.

**Pull request merged**